### PR TITLE
Added reservoir sampling to the pipeline

### PR DESCRIPTION
**Pull request by wfearn**

### CODE REVIEW

File: ankura/pipeline.py [hunk 1]
```diff
@@ -614,15 +615,44 @@ def build_docwords(corpus, V=None):
...
+    elif not num_train:
+        num_train = len(corpus.documents) - num_test
+    elif not num_test:
         num_test = len(corpus.documents) - num_train
 
-    doc_ids = np.random.permutation(len(corpus.documents))
-    train_ids, test_ids = doc_ids[:num_train], doc_ids[num_train: num_train+num_test]
-    train = Corpus([corpus.documents[d] for d in train_ids], corpus.vocabulary, corpus.metadata)
-    test = Corpus([corpus.documents[d] for d in test_ids], corpus.vocabulary, corpus.metadata)
+    print("Splitting test and train")
```
Comment (jefflund) [inline ankura/pipeline.py]: This is garbo bitmoji

File: ankura/pipeline.py [hunk 2]
```diff
@@ -614,15 +615,44 @@ def build_docwords(corpus, V=None):
...
+
+    try:
+        
+        doc_ids = np.random.permutation(len(corpus.documents))
+        train_ids, test_ids = doc_ids[:num_train], doc_ids[num_train: num_train+num_test]
+        train = Corpus([corpus.documents[d] for d in train_ids], corpus.vocabulary, corpus.metadata)
+        test = Corpus([corpus.documents[d] for d in test_ids], corpus.vocabulary, corpus.metadata)
+
+    #This occurs when your corpus is a doc stream instead of a list of documents.
```
Comment (jefflund) [inline ankura/pipeline.py]: More specifically, when the documents doesn't support random indexing. Be precise please. bitmoji

**Review by jefflund — COMMENTED** [review]
So close. Yet so far.

File: ankura/pipeline.py (hunk 1 above)
Comment (wfearn) [reply ankura/pipeline.py]: bitmoji

File: ankura/pipeline.py (hunk 2 above)
Comment (wfearn) [reply ankura/pipeline.py]: bitmoji

**Comment by jefflund** [comment]
bitmoji
bitmoji
bitmoji

**Pull request merged**